### DISCUSSION TITLE
Is it only me, but I want my clipboard to follow my mouse cursor

### DISCUSSION THREAD

**Discussion by kRHYME7** [prasanthrangan/hyprdots#161] [Ideas]
Is it only me, but I want my clipboard to follow my mouse cursor
For those who likes it that way. 
~/.config/hypr/scripts/cliphist.sh

# set rofi position

```

case $2 in
    1)  # top left
        pos="window {location: north west; anchor: north west; x-offset: 20px; y-offset: 20px;}"
        ;;
    2)  # top right
        pos="window {location: north east; anchor: north east; x-offset: -20px; y-offset: 20px;}"
        ;;
    3)  # bottom left
        pos="window {location: south east; anchor: south east; x-offset: -20px; y-offset: -20px;}"
        ;;
    4)  # bottom right
        pos="window {location: south west; anchor: south west; x-offset: 20px; y-offset: -20px;}"
        ;; 
         5) # Follow mouse cursor
        cursor_pos=$(hyprctl cursorpos)
        cursor_x=${cursor_pos%,*}
        cursor_y=${cursor_pos#*,}

        x_offset=10
        y_offset=-270
        cursor_x=$((cursor_x + x_offset))
        cursor_y=$((cursor_y + y_offset))
    # Construct the position string based on mouse cursor's position
    pos="window {location: north west; anchor: north west; x-offset: ${cursor_x}px; y-offset: ${cursor_y}px;}"

    ;;
esac
```

Changing offsets will  adjust  rofi clipboard position
    `pos="window {location:  0 ; anchor: north west; x-offset: ${cursor_x}px; y-offset: ${cursor_y}px;}"
`   Centers Rofi

**Comment by prasanthrangan** [comment]
Hi, thank you!
The clipboard already follows the mouse position, can you please check out the latest version

**Reply by kRHYME7** [reply]
My screen is 3000x2000

Yep I Have tried the latest commit but  it goes everywhere on my screen. The clipboard is far from the cursor on my case

(Have tried but I didn't try to modify the latest commit. I will try to check later what adjustments I need on my config.)

BTW This is the best set of dotfiles for Hyprland! The soon to be addition of eww will be 🔥🔥🔥

I will also insert my question:
 On caching themes/wallpaper/rofi does it consumes  ram? to make it faster to load? On what extent? Thank you !!!

**Reply by prasanthrangan** [reply]
Thank you so much!!

> My screen is 3000x2000
> Yep I Have tried the latest commit but it goes everywhere on my screen. The clipboard is far from the cursor on my case

ohh strange, what monitor is that??
ya thats is expected, it will reposition to a screen section based on cursor location, but not exactly to cursor location.

> BTW This is the best set of dotfiles for Hyprland! The soon to be addition of eww will be 🔥🔥🔥

I'm planning to integrate eww and option to switch between the bars on the fly... will do it later, as I'm stuck with work and exam till next month :(

> On caching themes/wallpaper/rofi does it consumes ram? to make it faster to load? On what extent? Thank you !!!

the wallpaper is cached once during install and its in here `~/.config/swww/.cache/` (about 60mb with default walls)...
but when you add new wallpapers, it will generate the required cache when you set the wallpaper for the first time, with small spike in cpu usage!
